ddtrace/llmobs/_experiment.py

```diff
@@ -128,8 +137,71 @@ def run(self, jobs: int = 1, raise_errors: bool = False, sample_size: Optional[i
         self._run_evaluators(task_results, raise_errors=raise_errors)
         return
 
-    def _run_task(self, jobs: int, raise_errors: bool = False, sample_size: Optional[int] = None) -> List[Any]:
-        return []
+    def _process_record(self, idx_record: Tuple[int, DatasetRecord]) -> Dict[str, JSONType]:
+        if not self._llmobs_instance or not self._llmobs_instance.enabled:
+            return {}
+        idx, record = idx_record
+        start_ns = time.time_ns()
+        with self._llmobs_instance._experiment(name=self._task.__name__, experiment_id=self._id) as span:
+            span_context = self._llmobs_instance.export_span(span=span)
+            if span_context:
+                span_id = span_context.get("span_id", "")
+                trace_id = span_context.get("trace_id", "")
+            else:
+                span_id, trace_id = "", ""
+            input_data = record["input_data"]
+            record_id = record.get("record_id", "")
```

Comment:
should we fail if there is no record_id? I'd rather hard fail for unexpected usage than propagating an error along

Reply:
Weird that record_id is not a required field on DatasetRecord, might have to look into this one

Reply:
@gary-huang do you know if record ID is a required field / should we fail if record ID is not provided? The base PR seems to indicate it is not required